test/unit/auth.js

```diff
@@ -62,10 +62,10 @@ describe('DropboxAuth', () => {
 
     it('throws an error if the redirect url isn\'t set and type is code', () => {
       const dbx = new Dropbox({ clientId: 'CLIENT_ID' });
-      chai.assert.equal(
-        dbx.auth.getAuthenticationUrl('', null, 'code'),
-        'https://www.dropbox.com/oauth2/authorize?response_type=code&client_id=CLIENT_ID',
-      );
+      dbx.auth.getAuthenticationUrl('', null, 'code')
+        .then((authUrl) => {
+          chai.assert.equal(authUrl, 'https://www.dropbox.com/oauth2/authorize?response_type=code&client_id=CLIENT_ID');
```

Comment:
Does this work? I thought we would need to do different logic for promises. Maybe just adding the done function would work here. But obviously if you fail, then it would fall into the promise reject not the main error handling